File: LC104_Maximum Depth of Binary Tree_Recursion.py
# -*- coding: utf-8 -*-
"""
Created on Wed Nov  4 13:33:17 2020

@author: Three
"""
 
# Definition for a binary tree node.
class TreeNode:
     def __init__(self, x):
         self.val = x
         self.left = None
         self.right = None
         
         
class Solution:
    def maxDepth(self, root):
        if root is None:
            return 0
        return max(self.maxDepth(root.left),self.maxDepth(root.right)) + 1
    
root = TreeNode(3)
root.left = TreeNode(10)
res = Solution()
print(res.maxDepth(root))

File: LC104_Maximum Depth of Binary Tree_Recursion.c
int maxDepth(struct TreeNode *root) {
    if (root == NULL) return 0;
    return fmax(maxDepth(root->left), maxDepth(root->right)) + 1;
}
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
